alternator/streams.cc

```diff
@@ -182,5 +182,277 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
     return make_ready_future<executor::request_return_type>(make_jsonable(std::move(ret)));
 }
 
+struct shard_id {
+    static constexpr auto marker = 'H';
+
+    utils::UUID table;
+    db_clock::time_point time;
+    cdc::stream_id id;
+
+    shard_id(utils::UUID tab, db_clock::time_point t, cdc::stream_id i)
+        : table(tab)
+        , time(t)
+        , id(i)
+    {}
+    shard_id(const sstring&);
+
+    friend std::ostream& operator<<(std::ostream& os, const shard_id& id) {
+        return os << marker << id.table << std::hex
```

Comment:
Because we only expect it to be used for lexical_cast? I am honestly on the fence on whether ostream operators should restore state, or never make assumptions on state to begin with. But maybe adding a stream state guard is prudent.

Reply:
What can I tell you, I never liked this operator<< stuff in C++ in the first place, and I'm not thrilled about bringing it into Alternator - but if it's stream.cc and you look after it (and take care not to print numbers in non-hex mode), it's fine with me :-)

Reply:
I've added flag preservation at least.